compiler/rustc_hir_typeck/src/lib.rs

```diff
@@ -421,6 +421,43 @@ fn report_unexpected_variant_res(
             err.multipart_suggestion_verbose(descr, suggestion, Applicability::MaybeIncorrect);
             err
         }
+        Res::Def(DefKind::Variant, _) if expr.is_none() => {
+            let fields = tcx
+                .expect_variant_res(res)
+                .fields
+                .raw
+                .iter()
+                .map(|field| field.name.to_ident_string())
+                .collect::<Vec<_>>();
+
+            err.span_label(span, format!("not a {expected}"));
+
+            let span = qpath.span().shrink_to_hi().to(span.shrink_to_hi());
+            err.span_suggestion_verbose(
+                span,
+                "use the struct variant pattern syntax",
+                if fields.is_empty() {
+                    " {}".to_string()
+                } else {
+                    format!(" {{ {} }}", fields.join(", "))
+                },
+                Applicability::MachineApplicable,
```

Comment:
These two suggestions cannot both be `MachineApplicable`. I would downgrade this down to `HasPlaceholders` if there are variants.

Reply:
Or honestly, it may be worth just suggesting `field: _, field: _` unconditionally (or `..` if there are.. e.g. > 4 fields), and adjusting the wording to note that they're being ignored.  Two suggestions seems a bit excessive...

Reply:
Alternatively, use [`span_suggestions`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/diagnostic/struct.Diag.html#method.span_suggestions) which supports MUTUALLY EXCLUSIVE suggestions. You'll need to make a single wording for both of them, though.

Reply:
OK. I fixed the diagnostic to just suggest a placeholder.